src/journal/journalctl.c

```diff
@@ -1572,6 +1583,22 @@ static int add_syslog_identifier(sd_journal *j) {
         return 0;
 }
 
+static int add_exclude_identifier(sd_journal *j) {
+        int r;
+
+        assert(j);
+
+        j->exclude = set_free(j->exclude);
+
+        STRV_FOREACH(i, arg_exclude_identifier) {
+                r = set_put_strdupv(&j->exclude, i);
+                if (r < 0)
+                        return r;
+        }
```

Comment:
`set_put_strdupv()` takes strv, so it is not necessary to run over `arg_exclude_identifier`. Simply pass it.
Also, please create a new set, then replace it.
```c
_cleanup_set_free_ Set *excludes = NULL;
int r;

assert(j);

r = set_put_strdupv(&excludes, arg_exclude_identifier);
if (r < 0)
        return r;

return set_free_and_replace(j->excludes, excludes);
```

Reply:
Thanks, changed the code that way